starboard/nplb/player_write_sample_test.cc

```diff
@@ -31,95 +33,114 @@ typedef SbPlayerTestFixture::VideoSamples VideoSamples;
+                 << "', key system '"
+                 << (key_system ? key_system : "no key system")
+                 << "', and output mode '"
+                 << (output_mode == kSbPlayerOutputModeDecodeToTexture
+                         ? "Decode To Texture"
+                         : "Punchout")
+                 << "'.";
   }
-
-  SB_LOG(INFO) << FormatString(
-      "Initialize SbPlayerWriteSampleTest with dmp file '%s' and with output "
-      "mode '%s'.",
-      audio_filename ? audio_filename : video_filename,
-      output_mode == kSbPlayerOutputModeDecodeToTexture ? "Decode To Texture"
-                                                        : "Punchout");
-}
-
-void SbPlayerWriteSampleTest::SetUp() {
-  ASSERT_NO_FATAL_FAILURE(
-      player_fixture_.reset(new SbPlayerTestFixture(GetParam())));
-}
-
-void SbPlayerWriteSampleTest::TearDown() {
-  ASSERT_NO_FATAL_FAILURE(player_fixture_.reset());
-}
+};
 
 TEST_P(SbPlayerWriteSampleTest, SeekAndDestroy) {
-  ASSERT_NO_FATAL_FAILURE(player_fixture_->Seek(kSbTimeSecond));
+  SbPlayerTestFixture player_fixture(GetParam());
+  EXPECT_FALSE(HasFailure());
+  ASSERT_NO_FATAL_FAILURE(player_fixture.Seek(kSbTimeSecond));
 }
 
 TEST_P(SbPlayerWriteSampleTest, NoInput) {
-  if (test_media_type_ == kSbMediaTypeAudio) {
-    ASSERT_NO_FATAL_FAILURE(player_fixture_->Write(AudioEOS()));
-  } else {
-    SB_DCHECK(test_media_type_ == kSbMediaTypeVideo);
-    ASSERT_NO_FATAL_FAILURE(player_fixture_->Write(VideoEOS()));
-  }
-  ASSERT_NO_FATAL_FAILURE(player_fixture_->WaitForPlayerEndOfStream());
+  SbPlayerTestFixture player_fixture(GetParam());
+  EXPECT_FALSE(HasFailure());
+  ASSERT_NO_FATAL_FAILURE(player_fixture.Write(
+      player_fixture.HasAudio() ? AudioEOS() : AudioSamples(),
+      player_fixture.HasVideo() ? VideoEOS() : VideoSamples()));
+  ASSERT_NO_FATAL_FAILURE(player_fixture.WaitForPlayerEndOfStream());
 }
 
 TEST_P(SbPlayerWriteSampleTest, WriteSingleBatch) {
-  int max_batch_size = SbPlayerGetMaximumNumberOfSamplesPerWrite(
-      player_fixture_->GetPlayer(), test_media_type_);
-  if (test_media_type_ == kSbMediaTypeAudio) {
-    ASSERT_NO_FATAL_FAILURE(
-        player_fixture_->Write(AudioSamples(0, max_batch_size).WithEOS()));
-  } else {
-    SB_DCHECK(test_media_type_ == kSbMediaTypeVideo);
-    ASSERT_NO_FATAL_FAILURE(
-        player_fixture_->Write(VideoSamples(0, max_batch_size).WithEOS()));
+  const int kMaxSamplesToWrite = 8;
+  int samples_to_write = kMaxSamplesToWrite;
+
+  SbPlayerTestFixture player_fixture(GetParam());
+  EXPECT_FALSE(HasFailure());
+
+  if (player_fixture.HasAudio()) {
+    samples_to_write = std::min(
+        samples_to_write, SbPlayerGetMaximumNumberOfSamplesPerWrite(
+                              player_fixture.GetPlayer(), kSbMediaTypeAudio));
   }
-  ASSERT_NO_FATAL_FAILURE(player_fixture_->WaitForPlayerEndOfStream());
+  if (player_fixture.HasVideo()) {
+    samples_to_write = std::min(
```

Comment:
Assume the audio batch size is 50, and video batch size is 4, now we are writing to both streams using a batch size of 4.  Shall we use 50 for audio and 4 for video?